doc/make.py

```diff
@@ -1,438 +1,423 @@
+        else:
+            msgstr = err_msg
+            subject += 'failed'
+
+        msg = MIMEText(msgstr)
+        msg['Subject'] = subject
+        msg['From'] = from_name
+        msg['To'] = to_name
+        server = smtplib.SMTP(server_str, port)
+        server.ehlo()
+        server.starttls()
+        server.ehlo()
+        server.login(login, pwd)
         try:
-            # remove stale file
-            os.remove('build/html/pandas.zip')
-        except:
-            pass
-
-
-def zip_html():
-    try:
-        print("\nZipping up HTML docs...")
-        # just in case the wonky build box doesn't have zip
-        # don't fail this.
-        os.system('cd build; rm -f html/pandas.zip; zip html/pandas.zip -r -q html/* ')
-        print("\n")
-    except:
-        pass
-
-def latex():
-    check_build()
-    if sys.platform != 'win32':
-        # LaTeX format.
-        if os.system('sphinx-build -j 2 -b latex -d build/doctrees '
-                     'source build/latex'):
-            raise SystemExit("Building LaTeX failed.")
-        # Produce pdf.
-
-        os.chdir('build/latex')
-
-        # Call the makefile produced by sphinx...
-        if os.system('make'):
-            print("Rendering LaTeX failed.")
-            print("You may still be able to get a usable PDF file by going into 'build/latex'")
-            print("and executing 'pdflatex pandas.tex' for the requisite number of passes.")
-            print("Or using the 'latex_forced' target")
-            raise SystemExit
-
-        os.chdir('../..')
-    else:
-        print('latex build has not been tested on windows')
-
-def latex_forced():
-    check_build()
-    if sys.platform != 'win32':
-        # LaTeX format.
-        if os.system('sphinx-build -j 2 -b latex -d build/doctrees '
-                     'source build/latex'):
-            raise SystemExit("Building LaTeX failed.")
-        # Produce pdf.
-
-        os.chdir('build/latex')
-
-        # Manually call pdflatex, 3 passes should ensure latex fixes up
-        # all the required cross-references and such.
-        os.system('pdflatex -interaction=nonstopmode pandas.tex')
-        os.system('pdflatex -interaction=nonstopmode pandas.tex')
-        os.system('pdflatex -interaction=nonstopmode pandas.tex')
-        raise SystemExit("You should check the file 'build/latex/pandas.pdf' for problems.")
-
-        os.chdir('../..')
-    else:
-        print('latex build has not been tested on windows')
-
-
-def check_build():
-    build_dirs = [
-        'build', 'build/doctrees', 'build/html',
-        'build/latex', 'build/plots', 'build/_static',
-        'build/_templates']
-    for d in build_dirs:
+            server.sendmail(from_name, to_name, msg.as_string())
+        finally:
+            server.close()
+
+    @classmethod
+    def html(cls):
+        """Build HTML documentation."""
+        cls._create_build_structure()
+        cls._sphinx_build('html')
+
+    @classmethod
+    def zip_html(cls):
```

Comment:
a lot of these are not used so can be removed

@TomAugspurger @jorisvandenbossche 

IOW we only need html, latex, clean, can remove all of the upload* ones

Reply:
zip_html is also used.
I did use the upload commands when I uploaded the docs. Not sure how Tom currently does it (I know we changed some things in where we upload the versioned docs and how they are symlinked)